uptrain/operators/language/sql.py

```diff
@@ -0,0 +1,185 @@
+"""
+Implement oeprators over sql data.
+"""
+
+from __future__ import annotations
+
+import itertools
+import json
+import os
+import typing as t
+
+import sqlglot
+from sqlglot.errors import ParseError
+from pydantic import BaseModel
+import polars as pl
+
+from uptrain.utilities.sql_utils import extract_tables_and_columns, extract_tables_and_columns_from_create, \
+    PLACEHOLDER_TABLE, execute_and_compare_sql
+
+if t.TYPE_CHECKING:
+    from uptrain.framework.base import *
+from uptrain.operators.base import *
+
+__all__ = ["HasStar", "GetSchemaDefinition", "ParseSQL", "ValidateTables", "ExecuteSQL"]
+
+
+# TODO: define a Table object and dump that into the dataframe
+class Table(BaseModel):
+    create_table_stmt: str
+    table_name: str
+    columns: list
+
+
+# operators to read schema definition from Spider sql dataset
+@register_op
+class GetSchemaDefinition(TableOp):
+    col_in_schema: str = "schema"
+    col_out: str = "schema_def"
+    col_out_tables: str = "schema_tables"
+    col_out_db_path: str = "db_path"
```

Comment:
No need for output column names. Check:

https://github.com/uptrain-ai/uptrain/blob/8d71e84d73dedf22948d4dbd7da77fb67b26d38d/uptrain/operators/language/grammar.py#L75

or https://github.com/uptrain-ai/uptrain/blob/8d71e84d73dedf22948d4dbd7da77fb67b26d38d/uptrain/operators/embs.py#L89

Reply:
columns names are not needed only if it is a ColumnOp